// dbmanager/index.js
var mysql = require('mysql');
var types = mysql.Types;

var typeNames = [];
for (var typeName in types) {
    typeNames[types[typeName]] = typeName;
}

var connections = [];
var database = '';

exports.connect = function () {
    var connection = mysql.createConnection({
        host: 'localhost',
        user: 'root',
        password: 'afrid123',
        // database: 'dating',
        dateStrings: true,
        multipleStatements: true
    });
    connection.connect();
    connections.push(connection);
    return connection;
}

exports.close = function () {
    connections[0].end();
}

exports.dataTypes = function () {
    return typeNames;
}

exports.connection = function (name) {
    return connections[0];
}

exports.getDatabases = function (callback) {
    connections[0].query('show databases', function (err, results) {
        var databases = [];
        results.forEach(function (item) {
            databases.push(item.Database);
        });
        callback(databases);
    });
}

exports.useDatabase = function (db, callback) {
    database = db;
    connections[0].query('use ' + db, function (err) {
        callback(err);
    });
}

exports.getTables = function (callback) {
    var tables = [];
    if (database.length == 0) {
        callback(tables);
        return;
    }
    connections[0].query('show tables', function (err, results) {
        results.forEach(function (item) {
            tables.push(item['Tables_in_' + database]);
        });
        callback(tables);
    });
}

exports.getDataView = function (table, callback) {
    var tables = [];
    if (database.length == 0) {
        callback(tables);
        return;
    }
    connections[0].query('select * from ' + table + ' limit 1000', function (err, results, fields) {
        callback(results, fields);
    });
}
